### PR TITLE
Bluetooth: controller: add INC_MOD() macro

### PR DESCRIPTION
**Pull request by alexeRadu**
This macros is an incrementation and modulo operation into one. It
replaces the "ticker_next_elapsed()" function and is also used in other
places ticker_start()/ticker_stop/etc functions to replace similar code.

This implementation does not add any performance penalty while reducing
the number of lines of code and adds clarity.

Signed-off-by: Radu Alexe <radu.alexe@nxp.com>

### CODE REVIEW

File: subsys/bluetooth/controller/ticker/ticker.c [hunk 1]
```diff
@@ -205,27 +205,12 @@ BUILD_ASSERT(sizeof(struct ticker_user_op) == TICKER_USER_OP_T_SIZE);
 #define TICKER_INSTANCE_MAX 1
 static struct ticker_instance _instance[TICKER_INSTANCE_MAX];
 
+#define INC_MOD(y, x) ((y) + 1 == (x) ? 0 : (y) + 1)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ticker/ticker.c]: Is there a Zephyr wide macro available for this that can be used? To be consistent, a zephyr wide use will be recommended.

File: subsys/bluetooth/controller/ticker/ticker.c [hunk 2]
```diff
@@ -583,9 +568,9 @@ void ticker_worker(void *param)
 	}
 
 	/* Queue the elapsed ticks */
-	if (instance->ticks_elapsed_first == instance->ticks_elapsed_last) {
-		ticker_next_elapsed(&instance->ticks_elapsed_last);
-	}
+	if (instance->ticks_elapsed_first == instance->ticks_elapsed_last)
+		instance->ticks_elapsed_last = INC_MOD(instance->ticks_elapsed_last, DOUBLE_BUFFER_SIZE);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ticker/ticker.c]: Enclose within {}, I am think its the convention zephyr-wide

File: subsys/bluetooth/controller/ticker/ticker.c [hunk 3]
```diff
@@ -1788,10 +1770,7 @@ u32_t ticker_update(u8_t instance_index, u8_t user_id, u8_t ticker_id,
 
 	user = &instance->users[user_id];
 
-	last = user->last + 1;
-	if (last >= user->count_user_op) {
-		last = 0U;
-	}
+	last = INC_MOD(user->last + 1, user->count_user_op);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ticker/ticker.c]: why additional + 1, the macro already has it. Same for other changes below

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ticker/ticker.c (hunk 1 above)
Comment (alexeRadu) [reply subsys/bluetooth/controller/ticker/ticker.c]: There is MODULO_INC() defined in 3 other modules but not exported. :D

File: subsys/bluetooth/controller/ticker/ticker.c (hunk 2 above)
Comment (alexeRadu) [reply subsys/bluetooth/controller/ticker/ticker.c]: I used checkpatch that I know complains if there are curly braces after simple if statements. But you are right, the zephyr documentation requires the use of braces even after simple if statements.

File: subsys/bluetooth/controller/ticker/ticker.c (hunk 3 above)
Comment (alexeRadu) [reply subsys/bluetooth/controller/ticker/ticker.c]: You are right. Nice catch!

**Pull request closed without merging**